schema_based_on_diagram.sql

```diff
@@ -0,0 +1,53 @@
+CREATE TABLE patients (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  name VARCHAR(100),
+  date_of_birth DATE,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_histories (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  admitted_at timestamp,
+  patient_id INT,
+  status VARCHAR(250),
+  FOREIGN KEY (patient_id) REFERENCES patients(id),
+  PRIMARY kEY (id)
+);
+
+CREATE TABLE invoices (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL,
+  generated_at timestamp,
+  payed_at timestamp,
+  medical_history_id INT,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+  PRIMARY kEY (id)
+);
+
+CREATE TABLE treatments(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  type VARCHAR(100),
+  name VARCHAR(100),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_histories_treatments(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  medical_histories_id INT,
+  treatments_id INT,
+  PRIMARY KEY (id),
+  FOREIGN KEY (medical_histories_id) REFERENCES medical_histories(id),
+  FOREIGN KEY (treatments_id) REFERENCES treatments(id)
+);
+
+CREATE TABLE invoice_items (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT,
+  PRIMARY kEY(id),
+  FOREIGN kEY (invoice_id) REFERENCES invoices(id),
```

Comment:
* Kindly create indexes for all foreign keys as stated in the project [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.) this is important because it helps improve performance by speeding up the process of retrieving the common reference values that the action will be performed on.